lib/loga/sidekiq6/job_logger.rb

```diff
@@ -10,22 +10,27 @@ def call(item, _queue)
 
         yield
 
-        with_elapsed_time_context(start) do
-          loga_log(
-            message: "#{item['class']} with jid: '#{item['jid']}' done", item: item,
-          )
-        end
+        ::Sidekiq::Context.current[:elapsed] = elapsed(start)
+
+        loga_log(message: "#{item['class']} with jid: '#{item['jid']}' done", item: item)
       rescue Exception => e # rubocop:disable Lint/RescueException
-        with_elapsed_time_context(start) do
-          loga_log(
-            message: "#{item['class']} with jid: '#{item['jid']}' fail", item: item,
-            exception: e
-          )
-        end
+        ::Sidekiq::Context.current[:elapsed] = elapsed(start)
+
+        loga_log(
+          message: "#{item['class']} with jid: '#{item['jid']}' fail", item: item,
+          exception: e
+        )
 
         raise
       end
 
+      def prepare(job_hash, &block)
+        super
+      ensure
+        # For sidekiq version < 6.4
+        Thread.current[:sidekiq_context] = nil
```

Comment:
@bliof-fc Thanks for opening this.

Can you explain how this keep things working with < 6.4.1 please?

Reply:
In the sidekiq commit this is introduced https://github.com/mperham/sidekiq/commit/01686c5e25dd83ab89c6d85b36feec26e203e81b#diff-ea165841c7daea83067e8903be22ed970caa75b69cf85d4e014691fcd827437cR41-R42

What actually changes is that, instead of using the `with(data) { whatever }` method https://github.com/mperham/sidekiq/blob/8ee13c703d569beba5b1c18484a991808b1e993a/lib/sidekiq/logger.rb#L8-L14 

We just add things to the context hash and reset it to nil when the job ends https://github.com/mperham/sidekiq/blob/46f4c6bd9ee0b2f5502a1dc0d03414989ad65063/lib/sidekiq/processor.rb#L123

So this line backports the cleanup